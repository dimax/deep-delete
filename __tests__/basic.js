const deepDelete = require('../src/commonjs');
const deepDeleteWorker = require('../src/experimental-workers');

const dataForOneKey = [
  {
    moreData: {
      keyToBeDeleted: 1,
    },
    keyToBeDeleted: 'Bye bye',
    anotherKey: 2,
  },
  1,
  null,
  'hola',
  new Date('2015-03-25T12:00:00Z'),
];
const dataForMultipleKeys = [
  {
    moreData: {
      key1ToDelete: 1,
    },
    key2ToDelete: 'Bye bye',
    anotherKey: 2,
  },
  1,
  null,
  'hola',
  new Date('2015-03-25T12:00:00Z'),
];

describe('Normal version', () => {
  test('Single key', () => {
    expect(deepDelete('keyToBeDeleted', dataForOneKey)).toMatchSnapshot();
  });
  test('Multiple keys', () => {
    expect(
      deepDelete(['key1ToDelete', 'key2ToDelete'], dataForMultipleKeys),
    ).toMatchSnapshot();
  });
});

describe('Worker version', () => {
  test('Single key', async () => {
    expect(
      await deepDeleteWorker('keyToBeDeleted', dataForOneKey),
    ).toMatchSnapshot();
  });

  test('Multiple keys', async () => {
    expect(
      await deepDeleteWorker(
        ['key1ToDelete', 'key2ToDelete'],
        dataForMultipleKeys,
      ),
    ).toMatchSnapshot();
  });
});
