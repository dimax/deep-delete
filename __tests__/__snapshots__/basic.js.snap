// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Normal version Multiple keys 1`] = `
Array [
  Object {
    "anotherKey": 2,
    "moreData": Object {},
  },
  1,
  null,
  "hola",
  2015-03-25T12:00:00.000Z,
]
`;

exports[`Normal version Single key 1`] = `
Array [
  Object {
    "anotherKey": 2,
    "moreData": Object {},
  },
  1,
  null,
  "hola",
  2015-03-25T12:00:00.000Z,
]
`;

exports[`Worker version Multiple keys 1`] = `
Array [
  Object {
    "anotherKey": 2,
    "moreData": Object {},
  },
  1,
  null,
  "hola",
  2015-03-25T12:00:00.000Z,
]
`;

exports[`Worker version Single key 1`] = `
Array [
  Object {
    "anotherKey": 2,
    "moreData": Object {},
  },
  1,
  null,
  "hola",
  2015-03-25T12:00:00.000Z,
]
`;
